activerecord/lib/active_record/core.rb

```diff
@@ -192,7 +192,9 @@ def find_by(*args) # :nodoc:
         hash = args.first
 
         return super if !(Hash === hash) || hash.values.any? { |v|
-          v.nil? || Array === v || Hash === v || Relation === v || Base === v
+          case v
+          when NilClass, Array, Range, Hash, Relation, Base then true
```

Comment:
This can be better written like:
```ruby
types = [NilClass, Array, Range, Hash, Relation, Base].freeze

v.class.in?(types)
```

it's a little bit odd to have a case statement which has only one branching and returns true.